ci-operator/step-registry/fips-check/node-scan/fips-check-node-scan-commands.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+set -e
+set -u
+set -o pipefail
+
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "setting the proxy"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+set_proxy
+run_command "oc whoami"
+run_command "oc version -o yaml"
+
+# skip test for disconnected clusters
+run_command "oc extract secret/pull-secret -n openshift-config --confirm --to /tmp"; ret=$?
+if [[ $ret -eq 0 ]]; then
+    auths=`cat /tmp/.dockerconfigjson`
+    if [[ $auths =~ "5000" ]]; then
+        echo "This is a disconnected env, skip it."
+        exit 0
```

Comment:
I'm not sure if this is not precise. How about https://github.com/openshift/openshift-tests-private/blob/e0154aa1be8d9340734c1c955b5dca77c0716dee/test/extended/networking/cloud_util.go#L1329-L1343 ?

Reply:
Yes, they are basically doing the same thing. The only  point is for a proxy cluster, the [checkDisconnect](https://github.com/openshift/openshift-tests-private/blob/e0154aa1be8d9340734c1c955b5dca77c0716dee/test/extended/networking/cloud_util.go#L1329-L1343) may not work as proxy needed.

Reply:
Updated. By the way, there is an error in https://github.com/openshift/openshift-tests-private/blob/e0154aa1be8d9340734c1c955b5dca77c0716dee/test/extended/networking/cloud_util.go#L1329-L1343, which didn't 't export proxy before curl, then it will skip a cluster with a backend proxy, too.